app/serializers/article/show_serializer.rb

```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
 class Article::ShowSerializer < ActiveModel::Serializer
-  attributes :id, :title, :body
+  attributes :id, :title, :body, :premium
+  
+  def body
+    current_user.nil? ? object.body[0..99] : object.body
```

Comment:
We're only checking if the user is logged in here, not if the article is premium. This will result in all articles being limited for visitors?

Reply:
good question

Reply:
yes, good pickup, changed the serializer method to display non premium articles now